libs/src/api/entityManager.ts

```diff
@@ -53,25 +66,226 @@ export class EntityManager extends Base {
+      responseValues.error = error
+      return responseValues
+    }
+  }
+
+  async unsavePlaylist(
+    playlistId: number
+  ): Promise<EntityManagerOperationResponse> {
+    const responseValues: EntityManagerOperationResponse =
+      this.getDefaultEntityManagerResponseValues()
+    try {
+      return await this.manageEntity({
+        userId: this.getCurrentUserId(),
+        entityType: EntityType.PLAYLIST,
+        entityId: playlistId,
+        action: Action.UNSAVE,
+        metadataMultihash: ''
+      })
+    } catch (e) {
+      const error = (e as Error).message
+      responseValues.error = error
+      return responseValues
+    }
+  }
+
+  async repostTrack(trackId: number): Promise<EntityManagerOperationResponse> {
+    const responseValues: EntityManagerOperationResponse =
+      this.getDefaultEntityManagerResponseValues()
+    try {
+      return await this.manageEntity({
+        userId: this.getCurrentUserId(),
+        entityType: EntityType.TRACK,
+        entityId: trackId,
+        action: Action.REPOST,
+        metadataMultihash: ''
+      })
+    } catch (e) {
+      const error = (e as Error).message
+      responseValues.error = error
+      return responseValues
+    }
+  }
+
+  async unrepostTrack(
+    trackId: number
+  ): Promise<EntityManagerOperationResponse> {
+    const responseValues: EntityManagerOperationResponse =
+      this.getDefaultEntityManagerResponseValues()
+    try {
+      return await this.manageEntity({
+        userId: this.getCurrentUserId(),
+        entityType: EntityType.TRACK,
+        entityId: trackId,
+        action: Action.UNREPOST,
+        metadataMultihash: ''
+      })
+    } catch (e) {
+      const error = (e as Error).message
+      responseValues.error = error
+      return responseValues
+    }
+  }
+
+  async repostPlaylist(
+    playlistId: number
+  ): Promise<EntityManagerOperationResponse> {
+    const responseValues: EntityManagerOperationResponse =
+      this.getDefaultEntityManagerResponseValues()
+    try {
+      return await this.manageEntity({
+        userId: this.getCurrentUserId(),
+        entityType: EntityType.PLAYLIST,
+        entityId: playlistId,
+        action: Action.REPOST,
+        metadataMultihash: ''
+      })
+    } catch (e) {
+      const error = (e as Error).message
+      responseValues.error = error
+      return responseValues
+    }
+  }
+
+  async unrepostPlaylist(
+    playlistId: number
+  ): Promise<EntityManagerOperationResponse> {
+    const responseValues: EntityManagerOperationResponse =
+      this.getDefaultEntityManagerResponseValues()
+    try {
+      return await this.manageEntity({
+        userId: this.getCurrentUserId(),
+        entityType: EntityType.PLAYLIST,
+        entityId: playlistId,
+        action: Action.UNREPOST,
+        metadataMultihash: ''
+      })
+    } catch (e) {
+      const error = (e as Error).message
+      responseValues.error = error
+      return responseValues
+    }
+  }
```

Comment:
All of these methods are really similar and could be simplified if we wanted - but I could go either way with explicitly writing out too
```
const createSocialMethod = (entityType: EntityType, action: Action) => async (entityId: number) => {
  return await this.manageEntity({ userId: this.getCurrentUserId(), entityType, entityId, action, metadataMultihash: '' })
}
async repostTrack = createSocialMethod(EntityType.Track, Action.REPOST)
async unrepostTrack = createSocialMethod(EntityType.Track, Action.UNREPOST)
async repostPlaylist = createSocialMethod(EntityType.Playlist, Action.REPOST)
async unrepostPlaylist = createSocialMethod(EntityType.Playlist, Action.UNREPOST)

```